### PR TITLE
Replace Filfox API with RPC Node

### PR DESCRIPTION
**Pull request by zediogoviana**
Why:
 - When the project started within the FVM Space Warp Hackathon, it depended on the Filfox API to request Events and Messages.
 - It's better to depend on an RPC Node. It gives us more flexibility and reliability. We can also now filter requests by Blocks we have already seen instead of needing to request all messages from the beginning.

How:
 - Using the Glif RPC endpoint.
 - Update the `Explorer` logic to adapt to this new requirement.
 - Replace all occurrences of `last_timestamp` to be `last_block`.

### CODE REVIEW

File: lib/dora.ex [hunk 1]
```diff
@@ -65,12 +65,12 @@ defmodule Dora do
...
 
-  def store_contract_information(address, timestamp, abi_path) do
-    :dets.insert(:addresses, {address, timestamp, abi_path})
+  def store_contract_information(address, block, abi_path) do
+    :dets.insert(:addresses, {address, block, abi_path})
 
     Dora.Contracts.create_or_update_contract(address, %{
       address: address,
-      last_timestamp: timestamp,
+      last_block: block,
```
Comment (DavideSilva) [inline lib/dora.ex:73]: Just double checking that you are not actually storing the entire block here and it's just the block number, right?

Comment (zediogoviana) [reply lib/dora.ex:73]: Nop, it's just the number. The function that calls this one does:

```
last_block =
  List.last(events)["blockNumber"]
  |> Utils.hex_to_int()
```

It just gets the number from the latest Event obtained.

Comment (DavideSilva) [reply lib/dora.ex:73]: That's what I figured. At first glance, was just confused by the name of the parameter being `block`

**Review by DavideSilva — APPROVED** [review]

File: lib/dora.ex (hunk 1 above)
Comment (zediogoviana) [reply lib/dora.ex:73]: I can change it to `block_number`. What do you think?

Comment (DavideSilva) [reply lib/dora.ex:73]: Might be best, actually. Just to increase readability of the function

Comment (zediogoviana) [reply lib/dora.ex]: Done. Check it now

**Pull request merged**